File: server/models/product.js
const mongoose = require("mongoose");

const productSchema = new mongoose.Schema({
    name: {
        type : String,
        required : true,
        trim: true,
    },
    description: {
        type : String,
        required : true,
        trim: true,
    },
    images: [
        {
            type : String,
            required : true,
            trim: true,
        },
    ],
    quantity : {
        type : Number,
        required : true,
    },
    price : {
        type : Number,
        required : true,
    },
    category : {
        type : String,
        required : true,
        trim: true,
    },
    //rating
});

const Product = mongoose.model("Product", productSchema);
module.exports = Product

File: lib/models/product/product.dart
import 'package:json_annotation/json_annotation.dart';

import '../rating/rating.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  final String name;
  final String description;
  final int price;
  final int quantity;
  final String category;
  final List<String> images;
  final String? id;
  final List<Rating>? ratings;

  Product({
    required this.name,
    required this.description,
    required this.price,
    required this.quantity,
    required this.category,
    required this.images,
    this.id,
    this.ratings,
  });

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);

  Map<String, dynamic> toJson() => _$ProductToJson(this);
}
